src/main/clojure/clara/rules/engine.cljc

```diff
@@ -1817,13 +1835,13 @@
                   (when (some-> node :production :props :no-loop)
                     (flush-updates *current-session*)))))
 
-            (recur (mem/next-activation-group transient-memory) next-group)))
+            (recur (mem/next-activation-group transient-memory) next-group flushed-updates-count)))
```

Comment:
Why no `inc` this time?'
Should it be set back to 0?

Reply:
My reasoning here was that this path is reached every time a :no-loop rule is fired, while the other paths where I incremented the count are only reached upon the end of an activation group.  I expect activation group switches to be much less numerous than individual rules firing, and I thought the risk of throwing in ultimately terminating cases here to be too high.

Reply:
Ok, that explanation makes sense. However, you may want to note that in a comment above this since I think it could be hard to understand looking at it later.

Reply:
Is this easier to understand now @mrrodriguez ? https://github.com/cerner/clara-rules/pull/425/commits/13ef0dfde268bd618092d34eb64f25edd2f00be8